cloud_platform/models/cloud_platform.py

```diff
@@ -0,0 +1,248 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+import logging
+import os
+import re
+
+from collections import namedtuple
+
+from distutils.util import strtobool
+
+from openerp import SUPERUSER_ID
+from openerp.osv import osv
+from openerp.tools.config import config
+
+
+_logger = logging.getLogger(__name__)
+
+
+def is_true(strval):
+    return bool(strtobool(strval or '0'.lower()))
+
+
+PlatformConfig = namedtuple(
+    'PlatformConfig',
+    'filestore'
+)
+
+
+class FilestoreKind(object):
+    db = 'db'
+    s3 = 's3'  # or compatible s3 object storage
+    swift = 'swift'
+    file = 'file'
+
+
+class CloudPlatform(osv.osv_abstract):
+    _name = 'cloud.platform'
+
+    def _platform_kinds(self):
+        # XXX for backward compatibility, we need this one here, move
+        # it in cloud_platform_exoscale in V11
+        return ['exoscale']
```

Comment:
work only for `exoscale` ?
Her we are not in 11 I don't understand the comment

Reply:
There was a project to have a generic `cloud_platform` module with extensions for each platform kind when migrating to v11 (but it never happened). So basically, `cloud_platform` is for exoscale and `cloud_platform_ovh` is an extension for ovh.

Reply:
Actually it happened in v12. Anyway, as it is a backport, no problem to have the same code than in 9.0, 10.0, 11.0